src/plots/mapbox/layout_attributes.js

```diff
@@ -0,0 +1,169 @@
+            ].join(' ')
+        },
+        y: {
+            valType: 'info_array',
+            role: 'info',
+            items: [
+                {valType: 'number', min: 0, max: 1},
+                {valType: 'number', min: 0, max: 1}
+            ],
+            dflt: [0, 1],
+            description: [
+                'Sets the vertical domain of this subplot',
+                '(in plot fraction).'
+            ].join(' ')
+        }
+    },
+
+    style: {
+        valType: 'string',
+        values: ['basic', 'streets', 'outdoors', 'light', 'dark', 'satellite', 'satellite-streets'],
+        dflt: 'basic',
+        role: 'style',
+        description: [
+            'Sets the Mapbox map style.',
+            'Either input the defaults Mapbox names or the URL to a custom style.'
+        ].join(' ')
+    },
+    center: {
+        lon: {
+            valType: 'number',
+            dflt: 0,
+            role: 'info',
+            description: 'Sets the longitude of the center of the map (in degrees East).'
+        },
+        lat: {
+            valType: 'number',
+            dflt: 0,
+            role: 'info',
+            description: 'Sets the latitude of the center of the map (in degrees North).'
+        }
+    },
+    zoom: {
+        valType: 'number',
+        dflt: 1,
+        role: 'info',
+        description: 'Sets the zoom level of the map.'
+    },
+    bearing: {
+        valType: 'number',
+        dflt: 0,
+        role: 'info',
+        description: 'Sets the bearing angle of the map (in degrees).'
+    },
+    pitch: {
+        valType: 'number',
+        dflt: 0,
+        role: 'info',
+        description: 'Sets the pitch angle of the map (in degrees).'
+    },
+
+    layers: {
+        _isLinkedToArray: true,
+
+        sourcetype: {
+            valType: 'enumerated',
+            values: ['geojson', 'vector'],
+            dflt: 'geojson',
+            role: 'info',
+            description: [
+                'Sets the source type for this layer.',
+                'Support for *raster*, *image* and *video* source types is coming soon.'
+            ].join(' ')
+        },
+
+        source: {
+            valType: 'any',
+            role: 'info',
+            description: [
+                'Sets the source data for this layer.',
+                'Source can be either a URL,',
+                'a geojson object (with `sourcetype` set to *geojson*)',
+                'or an array of tile URLS (with `sourcetype` set to *vector*).'
+            ].join(' ')
+        },
+
+        sourcelayer: {
+            valType: 'string',
+            dflt: '',
+            role: 'info',
+            description: [
+                'Specifies the layer to use from a vector tile source.',
+                'Required for *vector* source type that supports multiple layers.'
+            ].join(' ')
+        },
+
+        type: {
+            valType: 'enumerated',
+            values: ['line', 'fill'],
```

Comment:
This declaration is slightly controversial.

In `shapes` for example, we wouldn't make the distinction between fill shapes are not granted their own shape `type`: they are simple `rect` or `path` will fills.

But in the case of `mapbox` subplots, since not all GeoJSON feature sets can support fills, making the distinction between lines and fills felt more natural.